src/main/java/org/benetech/servicenet/service/impl/ActivityServiceImpl.java

```diff
@@ -55,66 +54,44 @@ public ActivityServiceImpl(OrganizationService organizationService, ConflictServ
     }
 
     @Override
-    public List<ActivityDTO> getAllOrganizationActivities(UUID systemAccountId) {
+    @Transactional(readOnly = true)
+    public Page<ActivityDTO> getAllOrganizationActivities(Pageable pageable, UUID systemAccountId) {
         List<ActivityDTO> activities = new ArrayList<>();
-        List<OrganizationDTO> orgs = organizationService.findAllWithOwnerId(systemAccountId);
+        PageRequest pageRequest = PageRequest.of(pageable.getPageNumber(), pageable.getPageSize());
+        Page<Organization> orgs = organizationService.findAllWithOwnerId(pageRequest, systemAccountId);
 
-        for (OrganizationDTO org : orgs) {
+        for (Organization org : orgs) {
             try {
-                Optional<ActivityDTO> activityOpt = getEntityActivity(org.getId(), org.getId());
+                Optional<ActivityDTO> activityOpt = getEntityActivity(org.getId());
                 activityOpt.ifPresent(activities::add);
             } catch (ActivityCreationException ex) {
                 log.error(ex.getMessage());
             }
         }
+        Comparator<ActivityDTO> comparator = ActivityComparatorFactory.createComparator(pageable);
```

Comment:
Sorting done after fetching batch from the db might result in inaccurately sorted records.